roles/common/templates/etc/ntp/cluster.conf

```diff
@@ -0,0 +1,4 @@
+# Maintain NTP peering with all other cluster nodes
+{% for host in groups['all'] -%}
+peer {{ hostvars[host][primary_interface]['ipv4']['address'] }}
```

Comment:
I really wish we had reliable DNS so that we didn't have to rely on host facts for things like this, and instead utilize inventory_hostname.